siac/walletcmd.go

```diff
@@ -489,6 +490,19 @@ func wallettransactionscmd() {
 
 // walletunlockcmd unlocks a saved wallet
 func walletunlockcmd() {
+	env_password := os.Getenv("SIA_WALLET_PASSWORD")
+	if len(env_password) > 0 {
+		qs := fmt.Sprintf("encryptionpassword=%s&dictonary=%s",
+			env_password, "english")
+		err := post("/wallet/unlock", qs)
+		if err != nil {
+			fmt.Println("Warning: SIA_WALLET_PASSWORD unlock failed")
+			fmt.Println("Trying interactive console input method next...")
+		} else {
+			fmt.Println("Wallet unlocked")
+			return
+		}
+	}
```

Comment:
As written, the warning will be printed every time you don't have the env var set, which sort of implies that the developers want you to set it. I'd rather it have it the other way around. How about:
```go
password := os.Getenv("SIA_WALLET_PASSWORD")
if password != "" {
	fmt.Println("Using SIA_WALLET_PASSWORD")
} else {
	var err error
	password, err = speakeasy.Ask("Wallet password: ")
	if err != nil {
 		die("Reading password failed:", err)
	}
}
qs := fmt.Sprintf("encryptionpassword=%s&dictonary=%s", password, "english")
```

Reply:
that's funny. i thought the notes-02.txt showed that it did not print the warning when the env var was not set.

Reply:
did it the long way due to taek comment ```taek - Today at 11:08 AM
just layering over the existing unlock command I think is fine
it'll check the env variable, and if that doesn't work / has errors, it'll ask for a password```